components/provisioner/internal/gardener/shoot_reconciler.go

```diff
@@ -264,3 +284,62 @@ func (r *ProvisioningOperator) setDeprovisioningFinished(cluster model.Cluster,
 
 	return nil
 }
+
+func setAuditConfig(shoot *gardener_types.Shoot, policyConfigMapName, subAccountId string) {
+	if shoot.Spec.Kubernetes.KubeAPIServer == nil {
+		shoot.Spec.Kubernetes.KubeAPIServer = &gardener_types.KubeAPIServerConfig{}
+	}
+
+	shoot.Spec.Kubernetes.KubeAPIServer.AuditConfig = &gardener_types.AuditConfig{
+		AuditPolicy: &gardener_types.AuditPolicy{
+			ConfigMapRef: &v12.ObjectReference{Name: policyConfigMapName},
+		},
+	}
```

Comment:
Correct me if I am wrong, but from what I understand the Config Map will not change during Shoot lifetime, so we can set it during the provisioning. I think it would be a little cleaner to not to modify spec in our controller.